packages/sources/chain-reserve-wallet/src/endpoint/wallet.ts

```diff
@@ -2,6 +2,19 @@ import { Requester, Validator } from '@chainlink/ea-bootstrap'
 import { AdapterResponse, Config, ExecuteWithConfig, InputParameters } from '@chainlink/types'
 import { ethers } from 'ethers'
 
+const networks = ['cardano', 'doge']
+const chainIds = ['mainnet', 'testnet']
+
+const networkChainMap: { [key: string]: number } = {
+  cardano_testnet: 0,
+  doge_testnet: 1,
+  cardano_mainnet: 2,
+  doge_mainnet: 3,
+}
+
+const isNetwork = (maybeNetwork: any) => networks.indexOf(maybeNetwork) !== -1
+const isChainId = (maybeChainId: any) => chainIds.indexOf(maybeChainId) !== -1
+
```

Comment:
With the changes to the [Validator input params](https://github.com/smartcontractkit/external-adapters-js/pull/1181), we should be able to specify this as an `options` attribute in `inputParameters`